site/src/xServices/users/usersXService.ts

```diff
@@ -132,261 +132,230 @@ export const usersMachine =
           updateUserRoles: {
             data: TypesGen.User
           }
-          getUserCount: {
-            data: TypesGen.UserCountResponse
-          }
         },
       },
       predictableActionArguments: true,
       id: "usersState",
-      type: "parallel",
+      on: {
+        UPDATE_FILTER: {
+          target: ".gettingUsers",
+          actions: ["assignFilter", "sendResetPage"],
+          internal: false,
+        },
+      },
+      initial: "startingPagination",
       states: {
-        count: {
-          initial: "gettingCount",
-          states: {
-            idle: {},
-            gettingCount: {
-              entry: "clearGetCountError",
-              invoke: {
-                src: "getUserCount",
-                id: "getUserCount",
-                onDone: [
-                  {
-                    target: "idle",
-                    actions: "assignCount",
-                  },
-                ],
-                onError: [
-                  {
-                    target: "idle",
-                    actions: "assignGetCountError",
-                  },
+        startingPagination: {
+          entry: "assignPaginationRef",
+          always: {
+            target: "gettingUsers",
+          },
+        },
+        gettingUsers: {
+          entry: "clearGetUsersError",
+          invoke: {
+            src: "getUsers",
+            id: "getUsers",
+            onDone: [
+              {
+                target: "idle",
+                actions: "assignUsers",
+              },
+            ],
+            onError: [
+              {
+                target: "idle",
+                actions: [
+                  "clearUsers",
+                  "assignGetUsersError",
+                  "displayGetUsersErrorMessage",
                 ],
               },
-            },
+            ],
           },
+          tags: "loading",
+        },
+        idle: {
+          entry: "clearSelectedUser",
           on: {
+            SUSPEND_USER: {
+              target: "confirmUserSuspension",
+              actions: "assignUserToSuspend",
+            },
+            DELETE_USER: {
+              target: "confirmUserDeletion",
+              actions: "assignUserToDelete",
+            },
+            ACTIVATE_USER: {
+              target: "confirmUserActivation",
+              actions: "assignUserToActivate",
+            },
+            RESET_USER_PASSWORD: {
+              target: "confirmUserPasswordReset",
+              actions: [
+                "assignUserIdToResetPassword",
+                "generateRandomPassword",
+              ],
+            },
+            UPDATE_USER_ROLES: {
+              target: "updatingUserRoles",
+              actions: "assignUserIdToUpdateRoles",
+            },
+            UPDATE_PAGE: {
+              target: "gettingUsers",
+              actions: "updateURL",
+            },
             UPDATE_FILTER: {
```

Comment:
you have two `UPDATE_FILTER`s. I was suggesting you delete this one, that's how the workspaces page works. But I'm not entirely sure which is better. Wherever it goes, `UPDATE_PAGE` should probably go in the same place. Also I didn't catch this before, but I don't think `UPDATE_FILTER` should have a target. Updating the filter sends a message to the pagination machine to reset the page, which sends a message back to this machine to go to `gettingUsers`, so we don't need to go there twice. (Statewise it doesn't matter, but I don't want it to kick off the fetch twice.)